### PR TITLE
EPMDJ-2880: Add plugin settings tab to service group

### PR DESCRIPTION
**Pull request by IliaSolovev**
Issue: EPMDJ-2880

### CODE REVIEW

File: src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx [hunk 1]
```diff
@@ -15,25 +15,26 @@ interface Props {
   className?: string;
   isOpen: boolean;
   onToggle: (arg: boolean) => void;
-  agentId: string;
+  dataId: string;
```
Comment (RomanAchilov) [inline src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx:18]: agentId

File: src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx [hunk 2]
```diff
@@ -15,25 +15,26 @@ interface Props {
...
+  agentType: 'service-group' | 'agent';
```
Comment (RomanAchilov) [inline src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx:19]: string

File: src/modules/plugins-settings-tab/plugins-settings-tab.tsx [hunk 3]
```diff
@@ -1,36 +1,37 @@
...
 import { AddPluginsModal } from './add-plugins-modal';
 import { NoPluginsStub } from './no-plugins-stub';
 
-import styles from './plugins-settings.module.scss';
+import styles from './plugins-settings-tab.module.scss';
 
 interface Props {
   className?: string;
-  agent: Agent;
+  data: Agent;
```
Comment (RomanAchilov) [inline src/modules/plugins-settings-tab/plugins-settings-tab.tsx:15]: agent

File: src/modules/plugins-settings-tab/plugins-settings-tab.tsx [hunk 4]
```diff
@@ -1,36 +1,37 @@
...
 
-const pluginsSettings = BEM(styles);
+const pluginsSettingsTab = BEM(styles);
 
-export const PluginsSettings = pluginsSettings(
+export const PluginsSettingsTab = pluginsSettingsTab(
   ({
     className,
-    agent: { id: agentId = '', plugins = [], buildVersion },
+    data,
```
Comment (RomanAchilov) [inline src/modules/plugins-settings-tab/plugins-settings-tab.tsx:23]: agent: { id: agentId = '', plugins = [], buildVersion },

File: src/pages/settings-page/agent-settings/agent-settings.tsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ export const AgentSettings = agentSettings(({ className, showMessage, agent }: P
     },
     {
       name: 'plugins',
-      component: <PluginsSettings agent={agent} />,
+      component: <PluginsSettingsTab data={agent} />,
```
Comment (RomanAchilov) [inline src/pages/settings-page/agent-settings/agent-settings.tsx:39]: agent

File: src/pages/settings-page/agent-settings/general-settings-form/general-settings-form.module.scss [hunk 6]
```diff
@@ -15,6 +15,9 @@
   }
 
   &__save-changes-button {
+    position: relative;
```
Comment (RomanAchilov) [inline src/pages/settings-page/agent-settings/general-settings-form/general-settings-form.module.scss:18]: ))

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx [hunk 7]
```diff
@@ -16,24 +16,25 @@ interface Props {
...
 
 export const AddPluginsModal = addPluginModal(({
-  className, isOpen, onToggle, agentId,
+  className, isOpen, onToggle, agentId, agentType,
 }: Props) => {
   const [selectedPlugins, setSelectedPlugins] = React.useState<string[]>([]);
   const { showMessage } = React.useContext(NotificationManagerContext);
-  const plugins = useWsConnection<Plugin[]>(defaultAdminSocket, `/agents/${agentId}/plugins`);
-  const handleLoadPlugins = loadPlugins(agentId, {
+  const connectionUrl = `/${agentType === 'agent' ? 'agents' : 'service-groups'}/${agentId}/plugins`;
```
Comment (pavyarov) [inline src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx:29]: This is not url. This is topic

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/modules/plugins-settings-tab/no-plugins-stub/no-plugins-stub.tsx [hunk 8]
```diff
@@ -6,15 +6,16 @@ import styles from './no-plugins-stub.module.scss';
 
 interface Props {
   className?: string;
+  children: string;
```
Comment (SMPlay) [inline src/modules/plugins-settings-tab/no-plugins-stub/no-plugins-stub.tsx:9]: maybe can you use `React.ReactNode` and in the future you can give here some content (e.g. `<div>Content</div>`)

File: src/modules/plugins-settings-tab/plugins-settings-tab.module.scss [hunk 9]
```diff
@@ -20,14 +20,19 @@
...
   &__header {
-    padding: 24px;
-    font-size: 18px;
+    display: flex;
+    justify-content: space-between;
+    height: auto;
+    padding: 24px 0;
+    margin: 0 24px;
+    font-size: 20px;
+    border-bottom: 1px solid map-get($monochrome, medium-tint);
```
Comment (SMPlay) [inline src/modules/plugins-settings-tab/plugins-settings-tab.module.scss:29]: up

File: src/modules/plugins-settings-tab/no-plugins-stub/no-plugins-stub.tsx (hunk 8 above)
Comment (IliaSolovev) [reply src/modules/plugins-settings-tab/no-plugins-stub/no-plugins-stub.tsx:9]: i  can use it but now it not necessary

File: src/modules/plugins-settings-tab/plugins-settings-tab.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+export const PluginsSettingsTab = pluginsSettingsTab(
+  ({
+    className,
+    agent: { id = '', buildVersion = '' },
+  }: Props) => {
+    const [isAddPluginOpen, setIsAddPluginOpen] = React.useState(false);
+    const { type: agentType } = useParams<{ type: 'service-group' | 'agent' }>();
+    const { push } = useHistory();
+    const pluginsTopic = `/${agentType === 'agent' ? 'agents' : 'service-groups'}/${id}/plugins`;
+    const plugins: Plugin[] = useWsConnection<Plugin[]>(defaultAdminSocket, pluginsTopic) || [];
```
Comment (RomanAchilov) [inline src/modules/plugins-settings-tab/plugins-settings-tab.tsx:32]: ??????

File: src/types/plugin.ts [hunk 11]
```diff
@@ -1,8 +1,8 @@
 export interface Plugin {
-  id?: string;
+  id: string;
```
Comment (RomanAchilov) [inline src/types/plugin.ts:2]: ?

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/modules/plugins-settings-tab/plugins-settings-tab.module.scss [hunk 12]
```diff
@@ -20,14 +20,19 @@
   }
 
   &__header {
-    padding: 24px;
-    font-size: 18px;
+    display: flex;
+    justify-content: space-between;
+    height: auto;
```
Comment (pavyarov) [inline src/modules/plugins-settings-tab/plugins-settings-tab.module.scss:25]: em?)  `auto` is default value

File: src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx [hunk 13]
```diff
@@ -15,23 +13,23 @@ interface Props {
   className?: string;
   isOpen: boolean;
   onToggle: (arg: boolean) => void;
-  agentId: string;
+  pluginsTopic: string;
```
Comment (pavyarov) [inline src/modules/plugins-settings-tab/add-plugins-modal/add-plugins-modal.tsx:16]: just want to metntion.
Passing topics via props was temprorary solution

**Review by pavyarov — CHANGES_REQUESTED** [review]

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

**Pull request merged**